### PR TITLE
[FEATURE] Support Hyperlinks to local pages in rst and md

### PR DESCRIPTION
**Pull request by linawolf**
local pages can be linked by source name, target name or pure name without ending. Links can be relative to the current file or absolute to the project root

### CODE REVIEW

File: packages/guides/src/ReferenceResolvers/PageHyperlinkResolver.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    public function resolve(LinkInlineNode $node, RenderContext $renderContext): bool
+    {
+        if (!$node instanceof HyperLinkNode) {
+            return false;
+        }
+
+        $canonicalDocumentName = $this->documentNameResolver->canonicalUrl($renderContext->getDirName(), $node->getTargetReference());
+        $canonicalDocumentName = rtrim($canonicalDocumentName, '.' . $renderContext->getOutputFormat());
```
Comment (jaapio) [inline packages/guides/src/ReferenceResolvers/PageHyperlinkResolver.php:37]: Hm, should this not be done on the parser side? Remove the extension from the file, as we have the extension there? This seems to be a complete list now, but it won't when somebody add's a custom format. Which is supported by the library we are creating.

File: tests/Integration/tests/markdown/link-page-md/input/subpages/index.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# Markdown with links
+
+This is a Markdown document with some basic formatting. 
+
+* [Subpage 1](subpage1.md)
+* [Page 1](../page1.md)
+* [Page 1](/page1.md)
+* [Page 1](/page1.html)
```
Comment (jaapio) [inline tests/Integration/tests/markdown/link-page-md/input/subpages/index.md:8]: I think this should be invalid, linking to an html page does not really make sense to me.

**Review by jaapio — CHANGES_REQUESTED** [review]

File: tests/Integration/tests/markdown/link-page-md/input/subpages/index.md (hunk 2 above)
Comment (linawolf) [reply tests/Integration/tests/markdown/link-page-md/input/subpages/index.md:8]: Mmh, wouldnt it make sense to like link to another output page? I think I have seen this used before?

File: packages/guides/src/ReferenceResolvers/PageHyperlinkResolver.php (hunk 1 above)
Comment (linawolf) [reply packages/guides/src/ReferenceResolvers/PageHyperlinkResolver.php:37]: I'll give it another look, that might be a good idea

**Comment by linawolf** [comment]
@jaapio please have another look

File: tests/Integration/tests/markdown/link-page-md/input/subpages/index.md (hunk 2 above)
Comment (linawolf) [reply tests/Integration/tests/markdown/link-page-md/input/subpages/index.md:8]: See for example: https://tomcam.github.io/least-github-pages/markdown-links.html

File: packages/guides/src/ReferenceResolvers/PageHyperlinkResolver.php [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
```
Comment (jaapio) [inline packages/guides/src/ReferenceResolvers/PageHyperlinkResolver.php:37]: Should this not be replaced by a `str_ends_with` and `substr`?

**Review by jaapio — APPROVED** [review]

**Comment by jaapio** [comment]
please rebase this branch so we can merge this. Thanks a lot for this improvement!

**Pull request merged**